hacker/models.py

```diff
@@ -14,92 +20,194 @@
+        - 'previous_attendant'
+        - 'tamu_student'
 
-    class Meta:         
-        abstract = True
+    Moved Fields: (moved TO `Application`)
+        - 'dietary_restrictions'
 
+    Altered Fields:
+        - 'date_submitted' -> 'application_date'
+
+'''
+
+
+
+class Application(models.Model):
+    major = models.CharField(max_length=50)
+    gender = models.CharField(choices=GENDERS, max_length=2)
+    classification = models.CharField(choices=CLASSIFICATIONS, max_length=2)
+    grad_year = models.IntegerField(choices=GRAD_YEARS, verbose_name='graduation year')          
+    dietary_restrictions = MultiSelectField(choices=DIETARY_RESTRICTIONS, verbose_name='dietary restrictions', blank=True)
+    travel_reimbursement_required = models.BooleanField(default=False) 
+
+    num_hackathons_attended = models.PositiveSmallIntegerField(default=0)
+    previous_attendant = models.BooleanField(default=False)
+    tamu_student = models.BooleanField(default=True)
 
-class Application(HackerProfile):
-    #resume = models.FileField( ... )           
     interests = models.TextField(max_length=200)
-    essay = models.TextField(max_length=200)
+    essay1 = models.TextField(max_length=200)
+    essay2 = models.TextField(max_length=200, null=True, blank=True)
+    essay3 = models.TextField(max_length=200, null=True, blank=True)
+    essay4 = models.TextField(max_length=200, null=True, blank=True)
+    notes = models.TextField(max_length=300, blank=True, help_text='Provide any additional notes and/or comments in the text box provide')
+    resume = models.FileField(upload_to='hacker_resumes', null=True, blank=True)
+    
     approved = models.NullBooleanField(blank=True)
-    date_approved = models.DateField(null=True, blank=True)         
+    queued_for_approval = models.NullBooleanField(blank=True)
+    
+    date_approved = models.DateField(null=True, blank=True)        
+    date_queued_for_approval = models.DateField(null=True, blank=True) 
     date_submitted = models.DateField(auto_now_add=True, blank=True)
-    notes = models.TextField(max_length=300, blank=True, help_text='Provide any additional notes and/or comments in the text box provide')
+    
+    hacker = models.OneToOneField(          
+        Hacker,
+        on_delete=models.CASCADE,
+    )
 
     def __str__(self):
-        return '%s, %s - Profile' % (self.hacker.last_name, self.hacker.first_name)
+        return '%s, %s - Application' % (self.hacker.last_name, self.hacker.first_name)
+
+    def get_first_name(self):
+        fn = getattr(self.hacker, 'first_name', None)
+        return fn
+
+    def get_last_name(self):
+        ln = getattr(self.hacker, 'last_name', None)
+        return ln
+
+    def get_email(self):
+        email = getattr(self.hacker, 'email', None)
+        return email
+
+    def get_is_active(self):
+        active = getattr(self.hacker, 'is_active', None)
+        return active
+
+    get_first_name.short_description = "First Name"
```

Comment:
I'm confused, what does this do? Could you link me to some documentation? It seems like you just added a property to a function?